### PR TITLE
Auth Secondary transitions

### PR DESCRIPTION
**Pull request by ozten**
- User already has a password
- User needs a password
- CLEANUP - network and user have a setPassword function, but
  there is no backend to support it.

### CODE REVIEW

**Comment by ozten** [comment]
Rebased against latest primary-transition-ux. @seanmonstar - using your set_password.ejs changes instead of mine.

File: lib/wsapi/address_info.js [hunk 1]
```diff
@@ -161,7 +161,8 @@ exports.process = function(req, res) {
     } else {
       r.state = STATE_TABLE[hasPassword][lastUsedAs][typeRightNow.type];
     }
-
+    r.state = 'transition_no_password';
```
Comment (seanmonstar) [inline lib/wsapi/address_info.js]: more sneaky ninjas

File: resources/static/dialog/js/modules/set_password.js [hunk 2]
```diff
@@ -36,6 +36,7 @@ BrowserID.Modules.SetPassword = (function() {
 
       self.renderDialog("set_password", {
         email: options.email,
+	idpName: helpers.getDomainFromEmail(options.email),
```
Comment (seanmonstar) [inline resources/static/dialog/js/modules/set_password.js]: already exists as `domain` a couple lines down. If you want this name for consistancy, just change the `domain` and usage in the template.

Comment (ozten) [inline resources/static/dialog/js/modules/set_password.js]: Missed this in the conflict resolution. I see set_password.ejs uses domain. Thanks! Removed.

File: resources/static/dialog/js/misc/state.js [hunk 3]
```diff
@@ -355,6 +364,13 @@ BrowserID.State = (function() {
         }
       }
       // Anything below this point means the address is a secondary.
+      else if ("transition_to_secondary" === info.state) {
+        startAction("doAuthenticate", info);
+      }
+      else if ("transition_no_password" === info.state) {
+        self.transitionNoPassword = info.email;
```
Comment (seanmonstar) [inline resources/static/dialog/js/misc/state.js:371]: I just piggybacked on the `resetPasswordEmail` for un-authed no_password. We should probably do the same in both cases tho.

**Pull request merged**